### PR TITLE
Prac 5 code review request

### PR DESCRIPTION
**Pull request by William-Bowman-JCU**
Can you review my code, @KennethNjeri

### CODE REVIEW

File: prac_05/state_names.py [hunk 1]
```diff
@@ -5,15 +5,17 @@
...
-while state_code != "":
-    if state_code in CODE_TO_NAME:
+for code in CODE_TO_NAME:
+    print(f'{code:3} is {CODE_TO_NAME[code]}')
+
+state_code = input("Enter short state: ").upper()
+while state_code != '':
+    try:
         print(state_code, "is", CODE_TO_NAME[state_code])
-    else:
```
Comment (KennethNjeri) [inline prac_05/state_names.py]: this looks misplaced

File: prac_05/state_names.py [hunk 2]
```diff
@@ -5,15 +5,17 @@
...
+    except KeyError:
         print("Invalid short state")
-    state_code = input("Enter short state: ")
-
+    state_code = input("Enter short state: ").upper()
```
Comment (KennethNjeri) [inline prac_05/state_names.py:21]: this is a repetitive  prompt

File: prac_05/state_names.py [hunk 3]
```diff
@@ -5,15 +5,17 @@
...
 # TODO: Reformat this file so the dictionary code follows PEP 8 convention
-CODE_TO_NAME = {"QLD":"Queensland", "NSW": "New South Wales", "NT" : "Northern Territory", "WA" : "Western Australia",
+CODE_TO_NAME = {"QLD": "Queensland", "NSW": "New South Wales", "NT": "Northern Territory", "WA": "Western Australia",
             "ACT": "Australian Capital Territory", "VIC": "Victoria", "TAS": "Tasmania"}
 print(CODE_TO_NAME)
 
-state_code = input("Enter short state: ")
-while state_code != "":
-    if state_code in CODE_TO_NAME:
+for code in CODE_TO_NAME:
```
Comment (KennethNjeri) [inline prac_05/state_names.py:12]: i think this should be indented inside the if statement

File: prac_05/hex_colours.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+HEX_COLOURS = {
+    'Absolute Zero': '#0048ba',
+    'Acid Green': '#b0bf1a',
+    'Alice Blue': 'f0f8ff',
```
Comment (KennethNjeri) [inline prac_05/hex_colours.py:4]: missing hash symbol

File: prac_05/hex_colours.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    'Amaranth': '#e52b50',
+    'Amber': '#ffbf00',
+    'Amethyst': '#9966cc',
+    'Antique White': '#faebd7',
+    'Apricot': '#fbceb1',
+    'Aqua': '#00ffff'
+}
+for colour in HEX_COLOURS:
+    print(f'{colour:16} - {HEX_COLOURS[colour]}') 
+print()
```
Comment (KennethNjeri) [inline prac_05/hex_colours.py:15]: prints a blank line

File: prac_05/word_occurrences.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+words = sorted(text.split())
+words_dict = {}
+for word in words:
+    if word in words_dict:
+        words_dict[word] += 1
+    else:
+        words_dict[word] = 1
+
+for word in words_dict:
+    print(f'{word:10} : {words_dict[word]}')
```
Comment (KennethNjeri) [inline prac_05/word_occurrences.py]: use code to find the max length instead of predetermined number

File: prac_05/word_occurrences.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+"""
+Word Occurrences
+Estimate: 20 minutes
+Actual:   15 minutes
+"""
+
+text = input('Text: ')
+words = sorted(text.split())
+words_dict = {}
```
Comment (KennethNjeri) [inline prac_05/word_occurrences.py]: maybe words_to_count is a better dictionary name

File: prac_05/emails.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+Emails
+Estimate: 5 minutes
+Actual:   5 minutes
+"""
+
+email = input('Email: ')
+names = {}
+while email != '':
+    name = email.split('@')[0].replace('.', ' ').title()
+    edit_name = input(f'Is your name {name}? (Y/n) ').lower()
```
Comment (KennethNjeri) [inline prac_05/emails.py]: is edit_name a good variable name here

**Comment by lindsaymarkward** [comment]
"initial commit" is just for the repo, not per prac/task.

**Pull request merged**